sql/variables/system_variables.go

```diff
@@ -762,6 +762,22 @@ var systemVars = map[string]sql.SystemVariable{
 		SetVarHintApplies: false,
 		Type:              types.NewSystemEnumType("event_scheduler", "ON", "OFF", "DISABLED"),
 		Default:           "ON",
+		NotifyChanged: func(scope sql.SystemVariableScope, value sql.SystemVarValue) {
+			convertedVal, _, err := value.Var.Type.Convert(value.Val)
+			if err == nil {
+				// TODO: need to update EventScheduler state at runtime if applicable
+				switch strings.ToLower(convertedVal.(string)) {
+				case "on", "1":
+					// need access to valid analyzer and ctx to call eventscheduler.TurnOnEventScheduler()
+				case "off", "0":
+					// need to call eventscheduler.TurnOffEventScheduler()
+				case "disabled":
+					// Err: Variable 'event_scheduler' can't be set to the value of 'disabled'
+				default:
+					// Err: Variable 'event_scheduler' can't be set to the value of 'no'
+				}
+			}
+		},
```

Comment:
Seems like we either need a function that allows us to access the running server/engine (we have this for Dolt, but not at the GMS layer) or we need to modify the `NotifyChanged` callback so that it takes in the engine. It doesn't look like `NotifyChanged` is very widely used, so that's my recommendation.